src/plugins/sorting-task/index.tsx

```diff
@@ -0,0 +1,254 @@
+import React from "react";
+import { Editor } from "slate";
+
+import { List } from "immutable";
+import { Block, Text } from "slate";
+import "./styles.css";
+
+export default function SortingTask() {
+  return {
+    changes: {
+    },
+    helpers: {},
+    components: {
+      SortingTaskNode
+    },
+    plugins: [RenderSortingTaskNode],
+  };
+}
+
+const onClickNewQuestionButton = (event: any, editor: Editor, node: Node) => {
+  event.preventDefault();
+  appendNewAnswer(editor, node);
+};
+
+const appendNewAnswer = (editor: any, node: any) => {
+  const newTerm = Block.create({
+    type: "sorting-task_term",
+    nodes: List([Text.create("")]),
+  });
+  const newQuestion = Block.create({
+    type: "sorting-task_solution",
+    nodes: List([Text.create("")]),
+  });
+
+  const newAnswer = Block.create({
+    type: "sorting-task_question",
+    nodes: List([
+      newTerm,
+      newQuestion,
+    ]),
+  });
+
+  const lastIndex = node.nodes.count();
+  return editor
+  .insertNodeByKey(node.key, lastIndex, newAnswer)
+  .moveToEndOfNode(newTerm);
+};
+
+const onClickNextQuestionButton = (event: any, editor: any, node: any) => {
+  event.preventDefault();
+  const prevValue = node.data.get("viewItem") || 0;
+  editor.setNodeByKey(node.key, { data: { viewItem: prevValue + 1 } })
+};
+
+const onClickResetQuizButton = (event: any, editor: any, node: any) => {
+  event.preventDefault();
+  editor.setNodeByKey(node.key, { data: { viewItem: 0 } })
+  node.nodes.forEach((childNode: any) => {
+    editor.setNodeByKey(childNode.key, { data: { viewState: false } })
+  });
+};
+
+const onClickToggleQuestionButton = (event: any, editor: any, node: any) => {
+  event.preventDefault();
+  const currentState = node.data.get("viewState") || false;
+  editor.setNodeByKey(node.key, { data: { viewState: !currentState } })
+};
+
+function SortingTaskNode(props: any) {
+  const { children, node, editor, readOnly, ...attributes} = props;
+  const currentChild = children[node.data.get("viewItem") || 0];
+  return (readOnly ?
+    (
+      <div>
+        <b>Sortieraufgabe (Schüleransicht)</b>
+        {currentChild}
+        <button
+          type="button"
+          onClick={event => {
+              onClickNextQuestionButton(
+                event,
+                editor,
+                node,
+              )
+            }
+          }
+        >
+          Weiter
+        </button>
+        <button
+          type="button"
+          onClick={event => {
+              onClickResetQuizButton(
+                event,
+                editor,
+                node,
+              )
+            }
+          }
+        >
+          Reset
+        </button>
+      </div>
+    ) : (
+      <div>
+        <b>Sortieraufgabe (Lehreransicht)</b>
+        <table {...attributes}>
```

Comment:
Directly nesting `tr`s inside of a `table` element throws an error. While I think the HTML specs do not require wrapping `thead`, `tbody` elements etc., React doesn’t seem to like it.

```html
<thead>
  <tr>
    <th>Begriff</th>
    <th>Musterlösung</th>
  </tr>
</thead>
<tbody>
  {children}
</tbody>
```